### ISSUE TITLE
[tree] Possible error in TLeaf::GetLeafCounter

### ISSUE THREAD

**Issue by linev** [root-project/root#7507]
There is code around line 269:
```
   TLeaf* leaf = (TLeaf*) GetBranch()->GetListOfLeaves()->FindObject(countname);
   if (leaf == 0) {
      // Try outside the branch:
      leaf = (TLeaf*) pTree->GetListOfLeaves()->FindObject(countname);
   }
   //if not found, make one more trial in case the leaf name has a "."
   if (!leaf && strchr(GetName(), '.')) {
      char* withdot = new char[strlen(GetName())+strlen(countname)+1];
      strcpy(withdot, GetName());
      char* lastdot = strrchr(withdot, '.');
      strcpy(lastdot, countname);
      leaf = (TLeaf*) pTree->GetListOfLeaves()->FindObject(countname);
      delete[] withdot;
      withdot = 0;
   }
```
Second if is useless - it again calls `leaf = (TLeaf*) pTree->GetListOfLeaves()->FindObject(countname);`
Seems to be, one wants to use here `withdot` as argument. 
If not - complete block can be removed

**Comment by pcanal** [comment]
> Second if is useless - it again calls leaf = (TLeaf*) pTree->GetListOfLeaves()->FindObject(countname);

The first call is on the branch list of leaves and the second is on the global (TTree) list of leaves, isn't it? (i.e. different lists).

**Comment by linev** [comment]
No, I mean last `if` block.
 It create new string, concatenate names, but does not use this string. And made exactly same call again.

**Comment by pcanal** [comment]
Ah .. you are right .. it is a bug
```
> leaf = (TLeaf*) pTree->GetListOfLeaves()->FindObject(countname);
< leaf = (TLeaf*) pTree->GetListOfLeaves()->FindObject(withdot);
```

**Comment by linev** [comment]
Should I provide PR?

**Comment by pcanal** [comment]
Please :)

**Comment by linev** [comment]
Fixed in 6.22, 6.24 and master branches
